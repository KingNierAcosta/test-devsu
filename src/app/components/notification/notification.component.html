<div class="notification" [ngClass]="type" [innerHTML]="message">
</div>
