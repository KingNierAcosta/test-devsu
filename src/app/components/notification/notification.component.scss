.notification {
  padding: 10px;
  border: 1px solid #ccc;
  border-radius: 4px;
  margin-bottom: 10px;
}

.success {
  background-color: #d4edda;
  color: #155724;
  border-color: #c3e6cb;
}

.error {
  background-color: #f8d7da;
  color: #721c24;
  border-color: #f5c6cb;
}
