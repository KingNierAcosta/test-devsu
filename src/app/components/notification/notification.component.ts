import { Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationType } from '../../models/product.model';

@Component({
  selector: 'app-notification',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './notification.component.html',
  styleUrl: './notification.component.scss'
})
export class NotificationComponent {

  @Input() message: string = '';
  @Input() type: NotificationType = 'success';

}
