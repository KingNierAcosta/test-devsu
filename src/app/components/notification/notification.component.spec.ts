import { ComponentFixture, TestBed } from '@angular/core/testing';

import { NotificationComponent } from './notification.component';
import { NotificationType } from '../../models/product.model';

describe('NotificationComponent', () => {
  let component: NotificationComponent;
  let fixture: ComponentFixture<NotificationComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [NotificationComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(NotificationComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  it('should display and error message', () => {
    const inputValue = 'Test Message';
    const notificationType: NotificationType = 'error';

    component.message = inputValue;
    component.type = notificationType;

    fixture.detectChanges();

    const compiled = fixture.nativeElement;

    expect(compiled.querySelector('.notification').textContent).toContain(
      inputValue
    );
    expect(compiled.querySelector('.notification').classList.value).toContain(
      notificationType
    );
  });
});
