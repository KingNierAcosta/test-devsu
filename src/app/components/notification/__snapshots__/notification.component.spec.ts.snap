// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationComponent should match the snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="notification success"
  />
</div>
`;
